iree/turbine/kernel/wave/runtime/CMakeLists.txt

```diff
@@ -13,72 +13,104 @@ if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
-execute_process(
-  COMMAND "${Python_EXECUTABLE}" -c "import torch; print(torch.__path__[0])"
-  OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE torch_PATH
-  COMMAND_ERROR_IS_FATAL ANY)
-
-# Add all directories in /opt/rocm/lib/cmake to CMAKE_PREFIX_PATH
-file(GLOB subdirectories "/opt/rocm/lib/cmake/*")
-foreach(subdir ${subdirectories})
-  if(IS_DIRECTORY ${subdir})
-    list(APPEND CMAKE_PREFIX_PATH "${subdir}")
-  endif()
-endforeach()
-
-include_directories(${torch_PATH}/include ${torch_PATH}/include/torch/csrc/api/include/)
-link_directories(${torch_PATH}/lib)
-
-execute_process(
-  COMMAND "${Python_EXECUTABLE}" -c "import torch; print(int(torch._C._GLIBCXX_USE_CXX11_ABI))"
-  OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE torch_cxx_abi
-  COMMAND_ERROR_IS_FATAL ANY)
-set(TORCH_CXX_FLAGS -D_GLIBCXX_USE_CXX11_ABI=${torch_cxx_abi})
-
-find_package(hip REQUIRED)
-add_definitions(-D__HIP_PLATFORM_AMD__ -DUSE_ROCM)
-
-# Build the core parts of nanobind once
-nanobind_build_library(nanobind SHARED)
-
-# Compile an extension library
+# Check if dependencies are available (passed from setup.py)
+if (NOT DEFINED NANOBIND_AVAILABLE)
+  set(NANOBIND_AVAILABLE OFF)
+endif()
+
+if (NOT DEFINED TORCH_CUDA_AVAILABLE)
+  set(TORCH_CUDA_AVAILABLE OFF)
+endif()
+
+# Determine if wave runtime should be enabled
+set(WAVE_RUNTIME_ENABLED TRUE)
+
+if (NOT NANOBIND_AVAILABLE)
+  set(WAVE_RUNTIME_ENABLED FALSE)
+  message(STATUS "nanobind not available. Disabling wave runtime functionality.")
+endif()
+
+if (NOT TORCH_CUDA_AVAILABLE)
+  set(WAVE_RUNTIME_ENABLED FALSE)
+  message(STATUS "PyTorch CUDA support not available. Disabling wave runtime functionality.")
+endif()
+
+if (NOT WAVE_RUNTIME_ENABLED)
+  message(STATUS "Wave runtime is disabled. Skipping build.")
+  return()
+endif()
+
+# Only proceed with nanobind setup if it's available
+if (NANOBIND_AVAILABLE)
+  execute_process(
+    COMMAND "${Python_EXECUTABLE}" -m nanobind --cmake_dir
+    OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE nanobind_ROOT
+    COMMAND_ERROR_IS_FATAL ANY)
+  find_package(nanobind CONFIG REQUIRED)
+endif()
+
 add_library(wave_runtime MODULE runtime.cpp)
 
-# .. and link it against the nanobind parts
-set(TORCH_LIBRARIES ${torch_PATH}/lib/libtorch.so ${torch_PATH}/lib/libc10.so)
-target_link_libraries(wave_runtime PRIVATE nanobind hip::host ${TORCH_LIBRARIES})
-set_target_properties(wave_runtime PROPERTIES LINK_WHAT_YOU_USE TRUE)
+# Only proceed with PyTorch setup and build if runtime is enabled
+if(WAVE_RUNTIME_ENABLED)
+  # Link against torch, using the libraries and include dirs in site-packages.
+  execute_process(
+    COMMAND "${Python_EXECUTABLE}" -c "import torch; print(torch.__path__[0])"
+    OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE torch_PATH
+    COMMAND_ERROR_IS_FATAL ANY)
+
+  # Add all directories in /opt/rocm/lib/cmake to CMAKE_PREFIX_PATH
+  file(GLOB subdirectories "/opt/rocm/lib/cmake/*")
```

Comment:
No hardcoded paths to ROCm. This will only work on an increasingly small footprint of devices and platforms.

Reply:
Any suggestions on what I can use instead?

Reply:
You can check what https://github.com/pytorch/pytorch/blob/main/cmake/public/LoadHIP.cmake does, though we've been reworking that. The main sections to look at are around `find_package(hip ...)`

Reply:
That does have hardcoded paths to ROCM though?